src/haddock/modules/analysis/alascan/scan.py

```diff
@@ -0,0 +1,591 @@
+                    if not resname:
+                        resname = line[17:20].strip()
+                    if atom_name in ATOMS_TO_BE_MUTATED:
+                        # mutate
+                        line = line[:17] + mut_resname + line[20:]
+                        mut_pdb_l.append(line)
+                else:
+                    mut_pdb_l.append(line)
+    try:
+        mut_id = f'{RES_CODES[resname]}{target_resnum}{RES_CODES[mut_resname]}'
+    except KeyError:
+        raise KeyError(f"Could not mutate {resname} into {mut_resname}.")
+    mut_pdb_fname = Path(
+        pdb_f.name.replace('.pdb', f'-{target_chain}_{mut_id}.pdb'))
+    with open(mut_pdb_fname, 'w') as fh:
+        fh.write(''.join(mut_pdb_l))
+    return mut_pdb_fname
+
+
+def add_delta_to_bfactor(pdb_f, df_scan):
+    """Add delta scores as b-factors.
+    
+    Parameters
+    ----------
+    pdb_f : str
+        Path to the pdb file.
+    df_scan : pandas.DataFrame
+        Dataframe with the scan results for the model
+    
+    Returns
+    -------
+    pdb_f : str
+        Path to the pdb file with the b-factors added.
+    """
+    tmp_pdb_f = pdb_f.replace('.pdb', '_bfactor.pdb')
+    max_b, min_b = df_scan["delta_score"].max(), df_scan["delta_score"].min()
+    out_pdb_l = []
+    with open(pdb_f, 'r') as fh:
+        for line in fh.readlines():
+            if line.startswith('ATOM'):
+                chain = line[21]
+                resnum = int(line[22:26])
+                norm_delta = 0.0
+                # extracting all the elements of df_scan such that
+                # chain = chain and res = resnum
+                df_scan_subset = df_scan.loc[
+                    (df_scan["chain"] == chain) & (df_scan["res"] == resnum)
+                    ]
+                if df_scan_subset.shape[0] > 0:
+                    delta = df_scan_subset["delta_score"].values[0]
+                    norm_delta = 100 * (delta - min_b) / (max_b - min_b)
+
+                delta_str = f"{norm_delta:.2f}".rjust(6, " ")
+                line = line[:60] + delta_str + line[66:]
+            out_pdb_l.append(line)
+    with open(tmp_pdb_f, 'w') as out_fh:
+        out_fh.write(''.join(out_pdb_l))
+    # move tmp_pdb_f to pdb_f
+    os.rename(tmp_pdb_f, pdb_f)
+    return pdb_f
+
+
+def calc_score(pdb_f, run_dir):
+    """Calculate the score of a model.
+    Parameters
+    ----------
+    pdb_f : str
+        Path to the pdb file.
+    run_dir : str
+        Path to the run directory.
+    Returns
+    -------
+    score : float
+        Haddock score.
+    vdw : float
+        Van der Waals energy.
+    elec : float
+        Electrostatic energy.
+    desolv : float
+        Desolvation energy.
+    bsa : float
+        Buried surface area.
+    """
+    cmd = f"haddock3-score {pdb_f} --full --run_dir {run_dir}"
+    p = subprocess.run(
+        shlex.split(cmd),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+        )
```

Comment:
Not sure about this part, you are making a (external) subprocess call to run another CLI which is inside the same code. Here instead you should get the function instead, something like `import haddock.clis.cli_score.main as calc_score`.

Reply:
I'll check if that's possible, if it is then it's definitely cleaner!